### ISSUE TITLE
Request: Alternate authentication - Google Oauth, Authelia, etc

### ISSUE THREAD

**Issue by Veeedka** [jfroment/seedbox#45]
Hey there @jfroment, 

This is a solid collection of tools! My only request for this is to set up some alternate authentication methods rather than basic HTTP auth - I've been trying to work through how to modify the collection of scripts to do this, however authentication is fairly deeply rooted in everything, so I'm getting a bit stuck. 

Thank you!

**Comment by jfroment** [comment]
Hello,

Yeah I agree, a more complete authentication system is missing from this project, and that's why the next version will include it.
In the meantime you could try the still work-in-progress branch ``sso`` in git.
In there, there is alpha support for Authelia and integration in script and config files.

What is working:
- SSO authentation using Authelia (field sso: true for any service in config.yaml)

What is missing:
- Sample Authelia config files, you have to provide them.
- Documentation and examples
- Bypasses for external apps, such as LunaSea api endpoints.
- Configuration automation
- Testing. There might be some issues.
- Configuration migration from simple http auth to SSO.

I plan to move forward with it this summer, but I also discovered Zitadel which seems more customizable so I will do some tests. I cannot garantee Authelia will be the solution this project will use.

Hope this helps, in the meantime do not hesitate to give some feedback on this early implementation if you feel there is need.

Regards

**Comment by Jemmay** [comment]
Any update on this?
Love your project, so i tried to look into integrating it myself, but decided to just wait in case you was about to release a new version. 👍

**Comment by CosmicWebCreator** [comment]
@jfroment I implemented authentik in the seedbox. Can I create a PR ? It uses the config file to add authentik. However, the user will have to configure the domain level redirect within authentik.

**Comment by jfroment** [comment]
Yes there have been some improvements lately, Authentik is already configured in the ``sso`` branch (I replaced Authelia which was lighter but maybe too opinionated), which I re-aligned with the ``dev`` one recently.
Basically, in the ``sso`` branch, a ``sso`` flag on any rule can be set to redirect SSO auth to Authentik, like this:
![image](https://github.com/jfroment/seedbox/assets/2869798/9761b2a3-cb00-41ca-a2ba-066bd4148fa6)

Of course you'll have to enable the Authentik service first.

Behind the scenes, an additional Traefik  router is added to the dynamic rules, to redirect to Authentik services behind it.
It is up to the user to add an outpost + provider and related configuration in Authentik itself,so basically what is missing is documentation, and a way to maybe semi-automate some rules (for example, skip sso for /api paths in sonarr or radarr to make them work in mobile apps, I'm thinking of LunaSea for example).

Any improvements and contributions are welcome in this branch, please do not hesitate to get onboard :)

**Comment by CosmicWebCreator** [comment]
Wow ! I just checked the branch and you pretty much did what I did! Good job! You even added Authelia! The only difference between your branch and mine is now Uptime Kuma and a usenet downloader. I think you're ahead of me with the sso since you have authelia and I don't! Let me know if you need my help for any further changes requested. I'll be pleased to help ;)! Good project going on here!

**Comment by mmornati** [comment]
Hey, if someone is interested and just because I spent a lot of time trying to understand how the things on SSO branch were working, I delivered the SSO branch (and not only, I merged the dev too, and some things I added on my side like uptime kuma) with a little bit of documentation.
https://github.com/mmornati/seedbox
There are 3 different SSO services but in the end, the one working is Authentik.
https://github.com/mmornati/seedbox/blob/master/doc/apps/authentik.md
